packages/suite/src/middlewares/wallet/storageMiddleware.ts

```diff
@@ -89,12 +90,25 @@ const storageMiddleware = (api: MiddlewareAPI<Dispatch, AppState>) => {
             ) {
                 const { account } = action.payload;
                 const device = findAccountDevice(account, selectDevices(api.getState()));
-                const historicRates = selectHistoricFiatRates(api.getState());
                 // update only transactions for remembered device
                 if (isDeviceRemembered(device)) {
-                    storageActions.removeAccountHistoricRates(account.key);
                     storageActions.removeAccountTransactions(account);
                     api.dispatch(storageActions.saveAccountTransactions(account));
+                }
+            }
+
+            if (
+                isAnyOf(
+                    transactionsActions.removeTransaction,
+                    updateTxsFiatRatesThunk.fulfilled,
+                )(action)
+            ) {
+                const { account } = action.payload;
```

Comment:
I think `account` is available in `action.meta.arg`. It does not need to be returned from thunk

Reply:
aha ok